compiler/src/dotty/tools/dotc/ast/untpd.scala

```diff
@@ -311,11 +312,14 @@ object untpd extends Trees.Instance[Untyped] with UntypedTreeInfo {
     val (tycon, targs) = tpt match {
       case AppliedTypeTree(tycon, targs) =>
         (tycon, targs)
-      case TypedSplice(AppliedTypeTree(tycon, targs)) =>
-        (TypedSplice(tycon), targs map (TypedSplice(_)))
       case TypedSplice(tpt1: tpd.Tree) =>
-        val tycon = tpt1.tpe.typeConstructor
-        val argTypes = tpt1.tpe.argTypesLo
+        var tp = tpt1.tpe.dealias
+        if (tpt1.isInstanceOf[AppliedTypeTree] && !tp.isInstanceOf[ValueType]) {
+          ctx.error(ex"$tp is not a class type", tpt.pos)
+          tp = defn.ObjectType
+        }
```

Comment:
I don't think you need this test if you do that:
```diff
diff --git a/compiler/src/dotty/tools/dotc/typer/Typer.scala b/compiler/src/dotty/tools/dotc/typer/Typer.scala
index 8a08be37b..3251f2735 100644
--- a/compiler/src/dotty/tools/dotc/typer/Typer.scala
+++ b/compiler/src/dotty/tools/dotc/typer/Typer.scala
@@ -516,7 +516,8 @@ class Typer extends Namer
           case TypeApplications.EtaExpansion(tycon) => tpt1 = tpt1.withType(tycon)
           case _ =>
         }
-        checkClassType(tpt1.tpe, tpt1.pos, traitReq = false, stablePrefixReq = true)
+        if (checkClassType(tpt1.tpe, tpt1.pos, traitReq = false, stablePrefixReq = true) eq defn.ObjectType)
+          tpt1 = TypeTree(defn.ObjectType)
 
         tpt1 match {
           case AppliedTypeTree(_, targs) =>
```

Reply:
I checked that it passed all the tests. I've pushed it. Thank you!